### PR TITLE
Upgrade to core 1.3.0

### LINKED ISSUE #3793: Terminating app due to uncaught exception 'RLMException', reason: 'flock() failed: Interrupted system call
## Goals

What do you want to achieve?

A normal user flow

## Expected Results

No crash

## Actual Results

App crash for a reason that a can't find any info about

## Steps to Reproduce

What are steps we can follow to reproduce this issue?

I don't know what triggered it, it seams it happend during a write transaction, but it doesn't happen consistently.

## Code Sample

    JVTRoom *room = [self getRoomFromId:roomId];

    if (room) {

        RLMRealm *realm = [RLMRealm defaultRealm];

        [realm beginWriteTransaction];

        room.isPresentedRoom = presented;

        [realm commitWriteTransaction];

    }

## Version of Realm and Tooling

Realm version: ?

Realm ( 1.0.1 )

Xcode version: ?

7.3.1

iOS/OSX version: ?

9.3.2

Dependency manager + version: ?

Cocoapods 1.0.0

### PR DESCRIPTION
**Pull request by tgoyne**
Fixes #3793.

### CODE REVIEW

**Comment by austinzheng** [comment]
👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,11 +7,19 @@ x.x.x Release notes (yyyy-MM-dd)
 
 ### Enhancements
 
-* Swift: A `write` block may now `throw`, reverting any changes already made in the transaction.
+* Swift: A `write` block may now `throw`, reverting any changes already made in
+  the transaction.
+* Reduce address space used when committing write transactions.
+* Significantly reduce the download size of prebuilt binaries and slightly
+  reduce the final size contribution of Realm to applications.
```
Comment (bdash) [inline CHANGELOG.md]: What accounted for the reduction in size contributed to applications?

**Comment by bdash** [comment]
👍

File: CHANGELOG.md (hunk 1 above)
Comment (tgoyne) [inline CHANGELOG.md]: The changes to util::terminate slightly reduced the linked size (but not by a very exciting amount).

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,11 +7,19 @@ x.x.x Release notes (yyyy-MM-dd)
...
 
 ### Bugfixes
 
 * Fix a crash when reading the shared schema from an observed Swift object.
+* Fix a crash when deleting all objects in a Realm while fast-enumerating query
+  results from that Realm.
+* Handle EINTR from flock() rather than crashing.
+* Fix incorrect behavior following a call to `[RLMRealm compact]`.
```
Comment (bmunkholm) [inline CHANGELOG.md]: Could we describe how users could have experienced this?

Comment (tgoyne) [inline CHANGELOG.md]: It's not actually public (although a few users have used it anyway) and we didn't get any reports of problems with it, so a wall of text about a problem that probably never effected any of the people reading this doesn't seem useful.

**Comment by jpsim** [comment]
👍

File: CHANGELOG.md (hunk 2 above)
Comment (bmunkholm) [inline CHANGELOG.md]: +1

**Comment by cmelchior** [comment]
We are investigating what looks like a Core sorting bug on Android right now, so you might want to hold merging this until we have verified it either way.

**Comment by tgoyne** [comment]
Added a changelog entry for the sort order change. IMO from our perspective it's a bug fix as it brings us closer to the platform's sort order, and as we've never specified anything about our sort order it's only a breaking change in the sense that every functional change ever is.

**Comment by cmelchior** [comment]
I am a bit puzzled how you can say that. You are replacing one broken implementation with another? I assume you had the same functionality as described here: https://realm.io/docs/java/latest/#sorting-and-querying-on-string. At least users had time to work around that. 

Changing that to something else that doesn't work might make sense, but right now the ramifications are not clear. Danish is a small language but what if e.g French or German is suddenly broken as well. As a minimum this should at least be a major version bump.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,11 +7,21 @@ x.x.x Release notes (yyyy-MM-dd)
...
+* Align the sort order for punctuation with the order used by obj-c/swift sort
```
Comment (cmelchior) [inline CHANGELOG.md]: Only having this is misleading IMO, because other things have most likely changed as well.

**Comment by tgoyne** [comment]
That section of the docs is either nonsense or incorrect, depending on how you interpret it. We only support sorting Latin Extended 2 in the sense that it produces a consistent result, but that's true of the entire unicode range. We certainly don't produce "correct" results when sorting them and never have, as that's impossible with a locale-insensitive sort.

**Comment by JadenGeller** [comment]
Any news on this merge? It seems like the conversation died down.

**Comment by tgoyne** [comment]
Core is going to switch back to the old sort order so we might as well just wait for that.

**Comment by bdash** [comment]
👍

**Pull request merged**